### ISSUE TITLE
Formal vs informal language

### ISSUE THREAD

**Issue by jtynkkynen** [projectfluent/fluent#300]
Is there a way to handle politeness or formality of language, like formal and informal? This is very typical feature in languages like Japanese and Korean.

**Comment by alerque** [comment]
You could handle this with attributes similar to some of the gender examples floating around. Is there some usage that would not be covered by attributes?

**Comment by jtynkkynen** [comment]
Thanks. I was thinking I could do that. I believe it covers all that is needed.

**Comment by alerque** [comment]
@jtynkkynen Do you happen to know if the Unicode CLDR data includes information about what languages have formal/informal variants for what pars of speech?

**Comment by jtynkkynen** [comment]
@alerque I don't think they have that. I do not know other languages than Korean and Japanese that have honorifics to the level of verb endings, but Tamil and Thai at least have some kind of honorifics.

**Comment by gandaro** [comment]
@jtynkkynen Do you have an example for a use case where you would want the translator to be able to provide both formal and informal versions?

In software translations to German, we usually decide on the tone we want to use at the beginning and use that consistently.

**Comment by jtynkkynen** [comment]
@gandaro If you have an app that greets the user, it might need to be in different levels of honorifics for an older user and a younger user. Also software that sends emails or messages to users might need to send them in different levels of speech.

**Comment by Pike** [comment]
One way to potentially hook this up is to add a `TONE()` function at the `Localization` class level, which you prime with metadata it needs to decide which tone to use for which language. I'd expect that function to return different values for different languages, and with cross-language fallback in Fluent, you might end up in a situation where that matters.

This is obviously delicate to implement on the top-level stack, as you can't re-use bindings between users, for example.

Also, the nature of this function is to return seemingly unbound values, so this is going to also challenge the product cycle on the localization tool side. And if your implementation wants to be as strictly typed as possible, this might be a blocker.

**Comment by rugk** [comment]
> Do you have an example for a use case where you would want the translator to be able to provide both formal and informal versions?

Actually, I know many software projects that are translated to German that offer both "German (du)" and "German (Sie)", i.e. a formal and informal version to users to select. The best example/one good one may be Nextcloud.

**Comment by zbraniecki** [comment]
As Axel suggested, in Fluent, this could be:

```
key = { TONE() ->
        [formal] ...
       *[informal] ...
    }
```

**Comment by jtynkkynen** [comment]
Yes, this could even handle the honorifics levels of Korean and Japanese which are a bit more complex than formal and informal.

**Comment by tukusejssirs** [comment]
Just to note: even Slovak, Czech, Hungarian, Italian, Spanish (and probably even Polish) distinguish between formal and informal communication to another person (singular _you_).

Some languages change only the pronoun _(you)_, others change even the verb suffix.
